### PR TITLE
Coverity Fixes

### PR DESCRIPTION
**Pull request by kingthorin**
* SpiderHtmlFormParser - CID 157242 Unused variable. Don't assign defaultValue in getDefaultTestValue` if it already has a value.
* SpiderHtmlFormParserUnitTest - Updated to account for new changes.
* FormNoDefaultValues.html - Removed radio buttons as one is always selected (representing a default value).

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
It looks like the SpiderHtmlFormParser modification is causing a unit test failure. Looking at the output "unknown" vs. an actual value it seems the new behavior is superior and the unit test should be updated to align. Can someone confirm or correct me if I'm wrong?

File: zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java [hunk 1]
```diff
@@ -347,26 +347,27 @@ private static void addAll(SortedSet<FormControl> formControls, List<Element> el
...
+                    Iterator<String> iterator = predefValues.iterator();
                     defaultValue = iterator.next();
+
+                    // If there are more values, don't use the first, as it usually is a "No select"
+                    // item
+                    if (iterator.hasNext()) {
+                        defaultValue = iterator.next();
+                    }
                 }
             }
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java]: After this add the `defaultValue = DEFAULT_EMPTY_VALUE` if still null.

File: zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java [hunk 2]
```diff
@@ -347,26 +347,27 @@ private static void addAll(SortedSet<FormControl> formControls, List<Element> el
...
 
-        } else {
-            defaultValue = values.get(0);
+        } else if (defaultValue == null) {
+            defaultValue = values.get(0) == null ? DEFAULT_EMPTY_VALUE : values.get(0);
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java]: This could be as before (`values.get(0)` is non-null).

**Comment by kingthorin** [comment]
It seems like the latest round of modifications help further:
* Expected defaultValue f though fieldAttributes are id male and value m?
```
    Expected: is <uri=http://example.com/, targetUri=http://example.org/post, fieldName=gender, defaultValue=f, values=[m, f], fieldAttributes={name=gender, id=male, type=radio, Control Type=RADIO, value=m}, formAttributes={action=http://example.org/post, method=POST, atta=valueA}>
```
* Now actually aligns defaultValue m, with fieldAttributes id male and value m. 
```
         but: was <uri=http://example.com/, targetUri=http://example.org/post, fieldName=gender, defaultValue=m, values=[m, f], fieldAttributes={name=gender, id=male, type=radio, value=m, Control Type=RADIO}, formAttributes={action=http://example.org/post, method=POST, atta=valueA}>
```
Unless I'm missing something that makes more sense.

The other two UnitTest failures are also now defined instead of unknown (as previously observed) [searching the Travis log for "unknown" and highlighting occurrences makes it stand out nicely.]

File: zap/src/test/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParserUnitTest.java [hunk 3]
```diff
@@ -1125,7 +1124,8 @@ public void shouldSetValuesToFieldsWithNoValueWhenParsingPostForm() {
         boolean completelyParsed = htmlParser.parseResource(msg, source, BASE_DEPTH);
         // Then
         assertThat(completelyParsed, is(equalTo(false)));
-        assertThat(listener.getNumberOfResourcesFound(), is(equalTo(9)));
+        assertThat(listener.getNumberOfResourcesFound(), is(equalTo(8)));
+        System.out.println(listener.getResourcesFound().get(5));
```
Comment (thc202) [inline zap/src/test/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParserUnitTest.java]: Debug?

Comment (kingthorin) [reply zap/src/test/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParserUnitTest.java]: Whoops, yup, will fix that tomorrow.

**Comment by kingthorin** [comment]
Leftover debug removed.

**Comment by thc202** [comment]
Thank you!

**Pull request merged**